synapse/lib/fifo.py

```diff
@@ -106,33 +106,46 @@ def flush(self):
         self.wind.flush()
         [wind.flush() for wind in self.winds]
 
+    def puts(self, items):
+        '''
+        Put a list of items into the Fifo.
+        This is a bulk access api for put().
+
+        Args:
+            items (list): A list of items to add
+        '''
+        with self.lock:
```

Comment:
Should we msgenpack all of the items before getting the lock and putting them into the fifo?